sympy/functions/combinatorial/numbers.py

```diff
@@ -1858,12 +1858,27 @@ def stirling(n, k, d=None, kind=2, signed=False):
 
 
 @cacheit
-def _nT(n, k):
+def _nT(n, k=None):
     """Return the partitions of ``n`` items into ``k`` parts. This
     is used by ``nT`` for the case when ``n`` is an integer."""
-    if k == 0:
-        return 1 if k == n else 0
-    return sum(_nT(n - k, j) for j in range(min(k, n - k) + 1))
+    if k is None:
+        return partition(n)
+    if k > n:
+        return 0
+    if 2*k >= n:
+        return partition(n - k)
+    numparts = 0
+    if n == k or k == 1:
+        numparts = 1
+    elif n < k or k == 0:
```

Comment:
Why check for `k > n` again?

How about putting this `k==0` test with the first `k > n` test?